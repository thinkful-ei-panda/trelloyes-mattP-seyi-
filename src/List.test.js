import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import List from './List'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe('<List />', () => {
  const props = {
    key: 'test_id',
    header: 'Card title',
    cards: [1, 2, 3],
  }

  it('should render without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<List />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('renders as expected', () => {
    const wrapper = renderer
      .create(<List name={props.cards[0]} />)
      .toJSON()
    expect(wrapper).toMatchSnapshot()
  })

  it('renders the complete form', () => {
    const wrapper = shallow(<List defaultProps />);
    wrapper.find('button').at(0).simulate('click');
    expect(toJson(wrapper)).toMatchSnapshot();
  })
})
