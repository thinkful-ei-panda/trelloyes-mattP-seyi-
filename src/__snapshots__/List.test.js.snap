// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> renders as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      Header
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <button
      className="List-add-button"
      onClick={[Function]}
      type="submit"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;
